src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -502,6 +502,56 @@ public void EnqueueRange(IEnumerable<TElement> elements, TPriority priority)
             }
         }
 
+        /// <summary>
+        /// Removes the first occurrence the equals the specified parameter.
+        /// </summary>
+        /// <param name="element">The element to look for.</param>
+        /// <param name="removedElement">The actual element that got removed from the queue.</param>
+        /// <param name="priority">The priority value associated with the removed element.</param>
+        /// <param name="equalityComparer">The equality comparer governing element equality.</param>
+        /// <returns><see langword="true"/> if matching entry was found and removed, <see langword="false"/> otherwise.</returns>
+        /// <remarks>
+        /// The method performs a linear-time scan of every element in the heap, removing the first value found to match the <paramref name="element"/> parameter.
+        /// In case of duplicate entries, what entry does get removed is non-deterministic, does not take priority into account and reflects the structure of the heap at the time of removal.
```

Comment:
```suggestion
        /// In case of duplicate entries, what entry does get removed is non-deterministic and does not take priority into account.
```
Might mentioning heap structure imply that given the same sequence of construction, what is removed might actually be deterministic? We might change the implementation.

Reply:
Good point, I'll remove that bit.